Expose more `Page` methods

johnvictor1965 opened the pull request:
Despite being implemented in `Target`, Puppeteer's `authenticate()` API is not available for regular `Page` users. This PR adds a new target message to port that. Same goes for `opener_id()`.

johnvictor1965 commented:
Closes #145

mattsse commented on src/handler/target.rs:
```diff
@@ -599,24 +603,6 @@ pub struct TargetConfig {
     pub cache_enabled: bool,
 }
 
-impl TargetConfig {
-    pub fn new(
-        ignore_https_errors: bool,
-        request_timeout: Duration,
-        viewport: Option<Viewport>,
-        request_intercept: bool,
-        cache_enabled: bool,
-    ) -> Self {
-        Self {
-            ignore_https_errors,
-            request_timeout,
-            viewport,
-            request_intercept,
-            cache_enabled,
-        }
-    }
-}
```
this is unrelated?

johnvictor1965 replied:
It is. My bad. I just suddenly started refactoring your crate at one point.

mattsse reviewed (approved):
lgtm,
one q re removed targetconfig function, 
I think this is probably unused but removing this would be a breaking change

johnvictor1965 commented:
I just added another one. And renamed `Target::opener()` to `Target::opener_id()`. Would that be a breaking change despite no one actually using `Target` in user code?

mattsse reviewed (approved):
this seems fine, I don't think this is used manually

johnvictor1965 commented:
Lemme just fix those imports for you
